rest/route/host_spawn.go

```diff
@@ -1128,53 +1129,75 @@ func (h *hostRunScript) Factory() gimlet.RouteHandler {
 
 func (h *hostRunScript) Parse(ctx context.Context, r *http.Request) error {
 	var err error
-	h.hostID, err = validateID(gimlet.GetVars(r)["host_id"])
-	if err != nil {
-		return errors.Wrap(err, "invalid host id")
-	}
-
-	hostScript := model.APIHostScript{}
-	if err = util.ReadJSONInto(util.NewRequestReader(r), &hostScript); err != nil {
+	hostScriptOpts := model.APIHostScript{}
+	if err = util.ReadJSONInto(util.NewRequestReader(r), &hostScriptOpts); err != nil {
 		return errors.Wrap(err, "can't read host command from json")
 	}
-	h.script = hostScript.Script
+	h.script = hostScriptOpts.Script
+	h.hostIDs = hostScriptOpts.Hosts
 
 	return nil
 }
 
 func (h *hostRunScript) Run(ctx context.Context) gimlet.Responder {
-	user := MustHaveUser(ctx)
-	host, err := h.sc.FindHostByIdWithOwner(h.hostID, user)
-	if err != nil {
-		return gimlet.MakeJSONInternalErrorResponder(errors.Wrap(err, "can't get host"))
-	}
-	if host.Status != evergreen.HostRunning {
-		return gimlet.MakeJSONInternalErrorResponder(errors.Errorf("can't run script on host with status '%s'", host.Status))
-	}
+	u := MustHaveUser(ctx)
+	type hostOutput struct {
+		hostID string
+		output string
+	}
+	hostsOutput := make(chan hostOutput, len(h.hostIDs))
+	wg := &sync.WaitGroup{}
+	wg.Add(len(h.hostIDs))
+	for _, hostID := range h.hostIDs {
+		go func(ctx context.Context, hostID string, u *user.DBUser) {
+			defer wg.Done()
+			host, err := h.sc.FindHostByIdWithOwner(hostID, u)
```

Comment:
Previously, there was a sanity check in `FindHostsInRange` for superusers, who seem to be able to get hosts that might not be theirs. Will calling this prevent the superuser from running the script on the host if it's not theirs?

Reply:
FindHostByIdWithOwner will [return an error](https://github.com/evergreen-ci/evergreen/blob/fee82dd5f586ae3bde123f4e3b85ec084c6630b2/rest/data/host.go#L388) if the user is not a superuser and they don't own the host.